client/my-sites/promote-post-i2/main.tsx

```diff
@@ -138,9 +139,37 @@ export default function PromotedPosts( { tab }: Props ) {
 
 	const translate = useTranslate();
 
+	const content = [
```

Comment:
Not to change in this PR, but it may be easier if we extract the logic for fetching products to a hook, and do the filtering/data modification there.

It would be easier to follow the code and maintain it in the future if we decide to move this logic to the DSP inside a custom endpoint.

Reply:
Thanks for the suggestiong, @sbarbosa! In sounds tempting to move all the logic to one place.

In the nearest future we will work on
- adding search and filtering for both campaigns and posts,
- implementing blaze for Jetpack (utilizing the Odyssey alike approach).

Probably that time it would be the best opportunity to move that into a hook and even to a custom endpoint. I've made a note about your suggestion.